packages/core/src/api/apis/definitions/featureFlags.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import ApiRef from '../ApiRef';
+import {
+  UserFlags,
+  FeatureFlagsRegistry,
+  FeatureFlagsRegistryItem,
+} from '../../app/FeatureFlags';
+
+/**
+ * The feature flags API is used to toggle functionality to users across plugins and Backstage.
+ *
+ * Plugins can use this API to register feature flags that they have available
+ * for users to enable/disable, and this API will centralize the current user's
+ * state of which feature flags they would like to enable.
+ *
+ * This is ideal for Backstage plugins, as well as your own App, to trial incomplete
+ * or unstable upcoming features. Although there will be a common interface for users
+ * to enable and disable feature flags, this API acts as another way to enable/disable.
+ */
+
+export enum FeatureFlagState {
+  NotEnabled = 0,
+  Enabled = 1,
+}
+
+export interface FeatureFlagsApi {
+  /**
+   * Store a list of registered feature flags.
+   */
+  registeredFeatureFlags: FeatureFlagsRegistryItem[];
+
+  /**
+   * Get a list of all feature flags from the current user.
+   */
+  getFlags(): UserFlags;
+
+  /**
+   * Get a list of all registered flags.
+   */
+  getRegisteredFlags(): FeatureFlagsRegistry;
```

Comment:
Why do we need this one and getFlags? O_o

Reply:
In my view they serve separate concerns. `getFlags` is purely around getting a user's state of feature flags. `getRegisteredFlags` is to get all plugins who want to register flags and make it possible to get all the available feature flags. 

Trying to shoehorn them into a single data structure I think might cause problems. It would be easy to fix if feature flags don't have any metadata. But I would like to add space for being able to see which plugins are registering which plugins, possible descriptions, etc for them in the future. If you don't think this is worth trying to make forwards-compatible, I can remove the metadata logic so we can stick to just one function for everything - `getFlags()`

Reply:
Maybe `getFlags` could be better as `getUserFlags`

Reply:
Yeah I agree it would be good to store some metadata with the flags, so something like this is good. Think this is fine for now and then we can look at how to connect the plugin output to the implementation of this API in a nicer way. With that we should be able to do validation earlier, and we could get rid of the FeatureFlagsRegistry and make it a simple array of flags instead.